proposals/0200-raw-string-escaping-revision-1.md

```diff
@@ -0,0 +1,454 @@
+
+In this design, there is only *one* variety of string literals. There is no special "raw" syntax. A string literal is either 
+
+* a sequence of characters surrounded by double quotation marks ("), or 
+* a string that spans several lines surrounded by three double quotation marks.
+
+These are examples of Swift string literals:
+
+```
+"This is a single line Swift string literal"
+
+"""
+    This is a multi line
+    Swift string literal
+    """
+```
+
+In this form, the revised string design acts exactly like any other string. You use escape character sequences including string interpolation exactly as you would today. A backslash escape character tells the compiler that a sequence should be interpolated, interpreted as an escaped character, or represent a unicode scalar. 
+
+Swift's escape sequences include:
+
+* The special characters `\0` (null character), `\\` (backslash), `\t` (horizontal tab), `\n` (line feed), `\r` (carriage return), `\"` (double quotation mark) and `\'` (single quotation mark) 
+* Arbitrary Unicode scalars, written as `\u{n}`, where *n* is a 1–8 digit hexadecimal number with a value equal to a valid Unicode code point
+* Interpolated expressions, introduced by `\(` and terminated by `)`
+
+### Expanding Delimiters
+
+Our design adds customizable string delimiters. You may pad a string literal with one or more `#` (pound, Number Sign, U+0023) characters:
+
+```
+"This is a Swift string literal"
+
+#"This is also a Swift string literal"#
+
+####"So is this"####
+```
+
+The number of pound signs at the start of the string (in these examples, zero, one, and four) must match the number of pound signs at the end of the string. `"This"`, `#"This"#`, and `##"This"##` represent identical string values:
+
+```
+static-string-literal -> " quoted-text " |
+   """ multiline-quoted-text """ |
+   # static-string-literal #
+```
+
+Adding a pound signs *changes the string literal delimiter* allowing you to "cook" a string and include unescaped double quotes:
+
+```
+#"She said, "This is dialog!""#
+// Equivalent to "She said, \"This is dialog!\""
+```
+
+If you do add a backslash, it is interpreted as an extra character. This string literal includes the backslash and both double quote marks inside the string delimiters (`#"` and `"#`):
+
+```
+#"A \"quote"."#
+```
+
+When you need to include `#"` (pound-quote) or `"#` (quote-pound) in your character sequence, adjust the number of delimiting pound signs. This need should be rare.
+
+### Customized Escape Delimiters
+
+This design uses an *escape delimiter*, that is a sequence of one or more characters to indicate the beginning of an escape character sequence, rather than a single escape character.  Like Swift today, the escape delimiter begins with a backslash (Reverse Solidus, U+005C), and is followed by zero or more pound signs (Number Sign, U+0023). An escape delimiter in a string literal must match the number of pound signs used to delimit either end of the string.
```

Comment:
The second sentence is phrased so that it sounds like it's saying that Swift currently uses pound signs in escape delimiters, which is not true. It might be better to phrase it as:

```
Like Swift today, the escape delimiter begins with a backslash (Reverse Solidus, U+005C), but it is
now followed by zero or more pound signs (Number Sign, U+0023).
```